cli/dts/lib.deno.unstable.d.ts

```diff
@@ -129,6 +129,70 @@ declare namespace Deno {
     speed: number | undefined;
   }
 
+  /** All possible types for interfacing with foreign functions */
+  export type NativeType =
+    | "void"
+    | "u8"
+    | "i8"
+    | "u16"
+    | "i16"
+    | "u32"
+    | "i32"
+    | "u64"
+    | "i64"
+    | "usize"
+    | "isize"
+    | "f32"
+    | "f64";
+
+  type NativeTypeToJsType<T> = T extends "void" ? void
+    : T extends
+      | "u64"
+      | "i64"
+      | "usize"
+      | "isize" ? bigint
+    : T extends
+      | "u8"
+      | "i8"
+      | "u16"
+      | "i16"
+      | "u32"
+      | "i32"
+      | "f32"
+      | "f64" ? number
+    : never;
+
+  type MapParametersToJsType<P extends NativeType[]> = {
+    [I in keyof P]: NativeTypeToJsType<P[I]>;
+  };
+
+  /** A foreign function as defined by its parameter and result types */
+  export interface ForeignFunction {
+    parameters: NativeType[];
+    result: NativeType;
+  }
+
+  /** A dynamic library resource */
+  export interface DynamicLibrary<S extends Record<string, ForeignFunction>> {
+    /** All of the registered symbols along with functions for calling them */
+    symbols: {
+      [K in keyof S]: (
+        args: MapParametersToJsType<S[K]["parameters"]>,
+      ) => NativeTypeToJsType<S[K]["result"]>;
+    };
+
+    close(): void;
+  }
+
+  /** **UNSTABLE**: new API
+   *
+   * Opens a dynamic library and registers symbols
+   */
+  export function dlopen<S extends Record<string, ForeignFunction>>(
```

Comment:
Since we have `allow-ffi`, it would be helpful for googling/doc searches if this function somehow included the string "ffi". Maybe rename "dlopen" to "openffi" ?

Reply:
`ffiopen` has a nice ring